test/deep-required.ts

```diff
@@ -89,6 +93,9 @@ function testDeepRequired() {
     >,
     Assert<IsExact<DeepRequired<readonly (number | null | undefined)[]>, readonly (number | null)[]>>,
     Assert<IsExact<DeepRequired<Array<number | null | undefined>>, Array<number | null>>>,
+    Assert<IsExact<DeepRequired<ReadonlyArray<number | null | undefined>>, ReadonlyArray<number | null>>>,
+    Assert<IsExact<DeepRequired<ExtendedArray<number | null | undefined>>, Array<number | null>>>,
```

Comment:
This case is slightly odd, in that types that extend Array get mapped back to `Array` as a side-effect of applying `DeepRequired` (and the same goes for types that extend `ReadonlyArray`).

This matches the existing behavior in deep-readonly, but it's less strange there, since the type is implicitly responsible for converting array types to `ReadonlyArray` types.

Reply:
this is a side-effect of conditional types that you have added: 

```ts
Type extends Array<Values>
    ? Array<Exclude<DeepRequired<Values>, undefined>>
    : ReadonlyArray<Exclude<DeepRequired<Values>, undefined>>
```

Now it's mapped back to `Array` or `ReadonlyArray` depending on what it `extends`